src/CameraComponent_V4L2.cpp

```diff
@@ -123,32 +129,40 @@ void CameraComponent_V4L2::initDefaultValues()
     saveParameter(CameraParameters::PIXEL_FORMAT, CameraParameters::ID_PIXEL_FORMAT_YUV420P);
 }
 
-int CameraComponent_V4L2::getParam(const char *param_id, char *param_value)
+int CameraComponent_V4L2::getParam(const char *param_id, char *param_value, size_t value_size)
 {
     // TODO :: Do appropriate checks
     // query the value set in the map and fill the output, return appropriate value
     std::string value = camParam.getParameter(param_id);
-    strcpy(param_value, value.c_str());
-    return 1;
+    if (!value.empty())
+        return 1;
+
+    if (value_size < value.size())
+        strncpy(param_value, value.c_str(), value_size);
+    else
+        strcpy(param_value, value.c_str());
```

Comment:
This is unsafe. You can cause a buffer overflow or write less bytes than you should. I would review all usages of strcpy.

Reply:
Fixed.